### PR TITLE
Tariff authorisation

### PR DESCRIPTION
**Pull request by pyotrritchie**

### CODE REVIEW

File: src/Service/Modules/TariffModule.cs [hunk 1]
```diff
@@ -38,8 +43,8 @@ private object GetTariffs()
         }
 
         private object GetTariff(int id)
```
Comment (richardiphillips) [inline src/Service/Modules/TariffModule.cs]: I think you'll need to add the privileges to the GetAll option too. That will allow the create button on the index page to be disabled where appropriate

File: src/Domain/AuthorisationService.cs [hunk 2]
```diff
@@ -1,31 +1,31 @@
...
+                case AuthorisedAction.TariffAdmin:
+                    return this.CanEditOrCreate(AuthorisedAction.TariffAdmin, privileges);
                 default:
                     return false;
-                    break;
             }
         }
 
-        private bool CanEditOrCreateVatCodes(IEnumerable<string> privileges)
+        private bool CanEditOrCreate(string action, IEnumerable<string> privileges)
```
Comment (richardiphillips) [inline src/Domain/AuthorisationService.cs]: I don't think you need this anymore. Could go straight to the satisfies method?

File: src/Domain/AuthorisationService.cs [hunk 3]
```diff
@@ -1,31 +1,31 @@
...
     using System.Collections.Generic;
     using System.Linq;
 
     public class AuthorisationService : IAuthorisationService
     {
-        public bool HasPermissionFor(AuthorisedAction action, IEnumerable<string> privileges)
+        public bool HasPermissionFor(string action, IEnumerable<string> privileges)
         {
             switch (action)
             {
```
Comment (richardiphillips) [inline src/Domain/AuthorisationService.cs]: In fact do you even need the switch anymore? Just call the satisfies method regardless of what they're trying to do.

File: src/Service/Modules/TariffModule.cs (hunk 1 above)
Comment (pyotrritchie) [reply src/Service/Modules/TariffModule.cs]: 👍

File: src/Domain/AuthorisationService.cs (hunk 2 above)
Comment (pyotrritchie) [reply src/Domain/AuthorisationService.cs]: 👍

File: src/Domain/AuthorisationService.cs (hunk 3 above)
Comment (pyotrritchie) [reply src/Domain/AuthorisationService.cs]: 👍

**Pull request merged**